modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/LegacyColumns.scala

```diff
@@ -160,6 +160,24 @@ object LegacyColumns {
         BadRow.LoaderIgluError(processor, BadRowFailure.LoaderIgluErrors(nel), BadPayload.LoaderPayload(event))
       }.toEither
 
+  /**
+   * Replaces any Json type with a String type
+   *
+   * This must be run after transformation, but before handling schema evolution. It is needed
+   * because the legacy loader used String columns where the v2 loader would create Json columns
+   */
+  def dropJsonTypes(v2Field: V2Field): V2Field = {
```

Comment:
I'm wondering, what happens if in `legacyTypeToV2Type` [here](https://github.com/snowplow-incubator/snowplow-bigquery-loader/blob/819f267ed3795adef0232e5de9e7727a8aaa8b54/modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/LegacyColumns.scala#L184) if we map to `V2Type.String` instead? So don't use `V2Type.Json` anywhere in legacy columns. What is then broken?

If we can  use `V2Type.String`, then do we still need to handle JSONs explicitly later (like here)?

Reply:
It's because the legacy ddl contains a String type under several circumstances:
1. The field is a really a string field
2. An object with no properties defined
3. An enum comprising numbers AND strings

Let's consider example 2.  If the json schema contains:

```
"a": {"type": "object"}
```

...and the event contains:

```
"a": {"xxx": "yyy"}
```

...and if we returned `V2Type.String` [on the line you said](https://github.com/snowplow-incubator/snowplow-bigquery-loader/blob/819f267ed3795adef0232e5de9e7727a8aaa8b54/modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/LegacyColumns.scala#L184), then the casting would fail [on this line](https://github.com/snowplow/schema-ddl/blob/414a2d76902b8653433154a1c7029b15768ed888/modules/core/src/main/scala/com.snowplowanalytics/iglu.schemaddl/parquet/Caster.scala#L67) so we'd get a failed event.

So I work around that problem by returning `V2Type.Json` and then handling the consequences later.  It's hacky, but it works.